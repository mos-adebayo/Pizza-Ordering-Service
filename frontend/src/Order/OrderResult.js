import React from 'react';
import PropTypes from 'prop-types';
import { Table, Badge } from "react-bootstrap";
import { appHelpers } from "../util/helper"
import Pagination  from "../util/pagination";

const OrderResult = ({ orders, activePage, pageLimit, totalOrders, handlePageClick }) => {
   const pageIndex = pageLimit * (activePage - 1);
    return (
        <>
            <Table striped bordered hover>
            <thead>
            <tr>
                <th>#</th>
                <th>Customer</th>
                <th>Pizza Type </th>
                <th>Pizza Size</th>
                <th>Quantity</th>
                <th>Status</th>
                <th>Address</th>
                <th>Date Ordered</th>
            </tr>
            </thead>
            <tbody>
            {
                orders.map((order, index) => <tr key={index}>
                    <td>{ (pageIndex) + (index + 1) }</td>
                    <td className={'text-capitalize'}>{`${order.customerName}`}</td>
                    <td>{`${order.pizzaType.name}`}</td>
                    <td>{`${order.pizzaSize.name}`}</td>
                    <td>{`${order.quantity}`}</td>
                    <td>
                        {
                            (order.status.name === "Pending") ? (
                                <Badge variant="warning">Pending</Badge>
                            ): (
                                <Badge variant="primary">{order.status.name}</Badge>
                            )
                        }
                    </td>
                    <td>{`${order.address}`}</td>
                    <td>{`${appHelpers.formatDate(order.createdAt)}`}</td>
                </tr>)
            }
            </tbody>
        </Table>
            
            <Pagination 
            activePage={activePage}
            totalItems={totalOrders} 
            handlePageClick={handlePageClick}
            pageLimit={pageLimit}
        />
        </>
    );
};

OrderResult.propTypes = {
    orders: PropTypes.array.isRequired
};

export default OrderResult;