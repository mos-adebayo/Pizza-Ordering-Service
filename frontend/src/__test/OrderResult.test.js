import React from 'react';
import {configure, mount } from 'enzyme';
import OrderResult from '../Order/OrderResult';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
configure({ adapter: new Adapter() });

function setup() {
    const props = {
        orders: [],
        activePage: 1,
        pageLimit: 5,
        totalOrders: 10,
        handlePageClick: jest.fn()
    };


    const enzymeWrapper = mount(
        <OrderResult {...props} />
    );
    return {
        props,
        enzymeWrapper,
    }
}
describe("Result grid component", () => {
    test('Result lists renders without crashing', () => {
        const  { props } = setup();

        const div = document.createElement('div');
        ReactDOM.render(
            <OrderResult {...props} />,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });

    test('Result lists snapshot', () => {
        const  { props } = setup();

        const component = renderer.create(
            <OrderResult {...props} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

    });


    test('should call page click handler callback when the page is clicked ', () => {
        const { enzymeWrapper, props } = setup();
        const target = enzymeWrapper.find('.page-item');
        expect(target.length).toBeGreaterThan(0);
        // target.at(1).simulate('click');
        // expect(props.handlePageClick.mock.calls.length).toBe(1)

    });

    test('page item should be 0 for no records ', () => {
        const { props } = setup();
        const  wrapper = mount(
            <OrderResult
                 {...props}
                 totalOrders={0}
            />
        );
        const target = wrapper.find('.page-item');
        expect(target.length).toBe(0);

    });

    test('page item should be calculated properly according to number of records and page limit ', () => {
        const { props } = setup();
        const  wrapper = mount(
            <OrderResult
                 {...props}
            />
        );
        const target = wrapper.find('.page-item');
        const numberOfPages = Math.ceil(props.totalOrders / props.pageLimit);
        expect(target.length).toBe(numberOfPages);

    });

});