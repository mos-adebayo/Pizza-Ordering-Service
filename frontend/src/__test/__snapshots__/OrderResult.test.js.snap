// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result grid component Result lists snapshot 1`] = `
Array [
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Customer
        </th>
        <th>
          Pizza Type 
        </th>
        <th>
          Pizza Size
        </th>
        <th>
          Quantity
        </th>
        <th>
          Status
        </th>
        <th>
          Address
        </th>
        <th>
          Date Ordered
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
  <ul
    className="pagination"
  >
    <li
      className="page-item active"
    >
      <span
        className="page-link"
        disabled={false}
        onClick={[Function]}
      >
        1
        <span
          className="sr-only"
        >
          (current)
        </span>
      </span>
    </li>
    <li
      className="page-item"
    >
      <a
        className="page-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        2
      </a>
    </li>
  </ul>,
]
`;
